### PR TITLE
Update to syn 0.15

### PR DESCRIPTION
**Pull request by dtolnay**
Release notes: https://github.com/dtolnay/syn/releases/tag/0.15.0

### CODE REVIEW

File: palette_derive/Cargo.toml [hunk 1]
```diff
@@ -14,7 +14,7 @@ license = "MIT OR Apache-2.0"
 proc-macro = true
 
 [dependencies]
-syn = { version = "0.14", features = ["visit", "extra-traits"] }
+syn = { version = "0.15.14", features = ["extra-traits"] }
```
Comment (Ogeon) [inline palette_derive/Cargo.toml]: I'm usually accepting patches, so I would have set it to just 0.15.

Comment (dtolnay) [reply palette_derive/Cargo.toml]: Done.

**Comment by Ogeon** [comment]
Great!

**Comment by Ogeon** [comment]
bors: r+

**Pull request merged**